modules/src/core/ics02_client/client_def.rs

```diff
@@ -150,7 +236,25 @@ pub trait ClientDef: Clone {
         port_id: &PortId,
         channel_id: &ChannelId,
         sequence: Sequence,
-    ) -> Result<(), Error>;
+    ) -> Result<(), Error> {
+        client_state.verify_height(height)?;
+        self.verify_delay_passed(ctx, height, connection_end)?;
+
+        let mut seq_bytes = Vec::new();
+        u64::from(sequence)
+            .encode(&mut seq_bytes)
+            .expect("buffer size too small");
+
+        let seq_path = SeqRecvsPath(port_id.clone(), *channel_id);
+        self.verify_membership(
```

Comment:
Note that `verify_membership()` doesn't provide access to the context, so (although this is only a default impl) it might be restrictive for light clients whose verification depends on data other than what we're giving this function here.